instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/pika_instrumentor.py

```diff
@@ -101,5 +94,24 @@ def _uninstrument(self, **kwargs: Dict[str, Any]) -> None:
                 channel._impl._consumers[key] = callback._original_callback
         PikaInstrumentor._uninstrument_channel_functions(channel)
 
+    def _decorate_channel_function(
+        self, tracer_provider: Optional[TracerProvider]
+    ) -> None:
+        self.original_channel_func = BlockingConnection.channel
+
+        def _wrapper(*args, **kwargs):
+            channel = self.original_channel_func(*args, **kwargs)
+            self.instrument_channel(channel, tracer_provider=tracer_provider)
+            return channel
+
+        BlockingConnection.channel = _wrapper
```

Comment:
Please use wrapt module for this like other instrumentation packages.

Reply:
Done